protected/humhub/modules/user/models/fieldtype/Birthday.php

```diff
@@ -41,90 +52,121 @@ public function rules()
+            Yii::$app->db->createCommand($query)->execute();
         }
 
+        // Delete the birthdate field (this is done by parent implementation)
         return parent::delete();
     }
 
     /**
-     * Saves this Profile Field Type
+     * @inheritdoc
      */
     public function save()
     {
+        // Add the birthdate_hide_year field
+        $columnNameHide = $this->profileField->internal_name . '_hide_year';
+        if (!Profile::columnExists($columnNameHide)) {
+            $query = Yii::$app->db->getQueryBuilder()->addColumn(Profile::tableName(), $columnNameHide, 'INT(1)');
+            Yii::$app->db->createCommand($query)->execute();
+        }
+
+        // Add the birthdate field
         $columnName = $this->profileField->internal_name;
-        if (!\humhub\modules\user\models\Profile::columnExists($columnName)) {
-            $query = Yii::$app->db->getQueryBuilder()->addColumn(\humhub\modules\user\models\Profile::tableName(), $columnName . '_hide_year', 'INT(1)');
+        if (!Profile::columnExists($columnName)) {
+            $query = Yii::$app->db->getQueryBuilder()->addColumn(Profile::tableName(), $columnName, 'DATE');
             Yii::$app->db->createCommand($query)->execute();
         }
 
         return parent::save();
     }
 
     /**
-     * Returns the Field Rules, to validate users input
-     *
-     * @param type $rules
-     * @return type
+     * @inheritdoc
      */
     public function getFieldRules($rules = array())
     {
-
-        $rules[] = [$this->profileField->internal_name . "_hide_year", 'in', 'range' => [0, 1]];
-        $rules[] = [$this->profileField->internal_name,
+        // Add validation for birthdate
+        $rules[] = [
+            $this->profileField->internal_name,
             \humhub\libs\DbDateValidator::className(),
             'format' => 'medium',
-            'convertToFormat' => null,
+            'convertToFormat' => 'Y-m-d',
             'max' => time(),
             'tooBig' => Yii::t('base', 'The date has to be in the past.')
         ];
+
+        // Add validation for birthdate_hide_year
+        $rules[] = [
+            $this->profileField->internal_name . '_hide_year',
+            'in',
+            'range' => [0, 1]
+        ];
+
         return parent::getFieldRules($rules);
     }
 
     /**
-     * Return the Form Element to edit the value of the Field
+     * @inheritdoc
      */
     public function getFieldFormDefinition()
     {
-        return [$this->profileField->internal_name => [
+        return [
+            $this->profileField->internal_name => [
                 'type' => 'datetime',
                 'format' => 'medium',
                 'class' => 'form-control',
                 'readonly' => (!$this->profileField->editable),
-                'yearRange' => (date('Y') - 100) . ":" . date('Y')
+                'yearRange' => (date('Y') - 100) . ":" . date('Y'),
+                'dateTimePickerOptions' => array(
```

Comment:
It's not a big deal but using `[]` rather than `array()` would make for better code consistency.